doc/build-unix.md

```diff
@@ -164,15 +163,13 @@ turned off by default.  See the configure options for upnp behavior desired:
 
 Berkeley DB
 -----------
-It is recommended to use Berkeley DB 4.8. If you have to build it yourself,
-you can use [the installation script included in contrib/](/contrib/install_db4.sh)
-like so:
-
-```shell
-./contrib/install_db4.sh `pwd`
-```
 
-from the root of the repository.
+BerkeleyDB 4.7 through 5.3 (inclusive) can be used. Most linux distributions ship BerkeleyDB 5.3.
+The `wallet.dat` files will be compatible as the database file format does not change within these versions.
+However the database transaction log files do have incompatible formats. But these log files are only relevant in unclean wallet closes.
+If a wallet is cleanly closed, the log files will be removed. If it is not cleanly closed, then a
+build using the same version of BDB will need to be used to open and close that wallet to guarantee that
```

Comment:
> If it is not cleanly closed, then a
> build using the same version of BDB will need to be used to open and close that wallet to guarantee that

What exactly happens in this case? Is there a clear error aborting the wallet load and saying there's a version discrepancy? Or is the wallet allowed to load with incomplete data? If it's the first, then this PR seems good. If it's the second, then allowing a case of data loss to avoid the need to type `--with-incompatible-bdb` doesn't seem like a too appealing tradeoff.

Another thing to think about (I don't this is a big concern if error handling above is adequate) is that "these log files are only relevant in unclean wallet closes" happens to be true right now. But it wasn't always true in previous releases, and it may not be true again in the future if we want to do something to improve large wallet performance without sacrificing data integrity.

But my main question is just whether there's an error or data loss a when 4.8 wallet software tries to open a wallet with 5.3 logs

Reply:
> Or is the wallet allowed to load with incomplete data?

The wallet is allowed to load. The logs are moved to a backup directory and an error is logged to both the debug.log and db.log. Unless the user is actively monitoring one of those files, they wouldn't know that there is an error and potential data loss.